gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineErrorProne.java

```diff
@@ -91,6 +102,48 @@ public void apply(Project project) {
                                             .setBootClasspath(new LazyConfigurationList(errorProneFiles)));
                         });
             }
+
+            // Create compile*Apply tasks that apply the errorprone-generated patch files
+            Task applyTask = project.getTasks().create(APPLY_TASK_NAME);
+            List<JavaCompile> compileTasks = ImmutableList.copyOf(
+                    project.getTasks().withType(JavaCompile.class).iterator());
+            compileTasks.forEach(javaCompile -> {
+                File patchFile = project.getBuildDir().toPath()
+                        .resolve("errorprone")
+                        .resolve(javaCompile.getName())
+                        .resolve("error-prone.patch")
+                        .toFile();
+                javaCompile.getOutputs().file(patchFile);
+
+                // TODO(gatesn): How can we flag this on/off? It appears to come with a hefty perf hit
+                ((ExtensionAware) javaCompile.getOptions()).getExtensions()
+                        .configure(ErrorProneOptions.class, errorProneOptions -> {
+                            // TODO(gatesn): Is there a way to discover error-prone checks?
+                            // Maybe service-load from a ClassLoader configured with annotation processor path?
+                            // https://github.com/google/error-prone/pull/947
+                            errorProneOptions.getErrorproneArgumentProviders().add(() -> ImmutableList.of(
+                                    "-XepPatchChecks:" + Joiner.on(',')
+                                            .join(errorProneExtension.getPatchChecks().get()),
+                                    "-XepPatchLocation:" + patchFile.getParentFile().getAbsolutePath()));
```

Comment:
could use in place application to directly apply fixes similar to `spotlessApply`

```suggestion
                                    "-XepPatchLocation:IN_PLACE"));
```